### PR TITLE
fix(lexer): print whitespace warning for \x0c

### LINKED ISSUE #108275: string continue non-ASCII whitespace confusing as `\x0C` is ASCII whitespace
### Code

```Rust
fn main() {
    assert!('\x0c'.is_ascii_whitespace());
    println!("Hello,\
         world!");
}
```

### Current output

```Shell
warning: non-ASCII whitespace symbol '\u{c}' is not skipped
 --> src/main.rs:3:21
  |
3 |       println!("Hello,\
  |  _____________________^
4 | |
               world!");
  | |      ^- non-ASCII whitespace symbol '\u{c}' is not skipped
  | |______|
  |
```

### Desired output

```Shell
warning: non-ASCII whitespace symbol '\u{c}' is not skipped
 --> src/main.rs:3:21
  |
3 |       println!("Hello,\
  |  _____________________^
4 | |
               world!");
  | |      ^- '\u{c}' is not skipped, only `\n`, `\r`, `\t` and ` ` are skipped
  | |______|
  |
```

### Rationale and extra context

According to `char::is_ascii_whitespace`, `\x0c` is ASCII whitespace.

### Other cases

_No response_

### Anything else?

Not sure if this is actually an issue, just an inconsistency I found.

### PR DESCRIPTION
**Pull request by bvanjoi**
- close https://github.com/rust-lang/rust/issues/108275
- discussion: https://github.com/rust-lang/rust/pull/108403

### CODE REVIEW

File: compiler/rustc_parse/locales/en-US.ftl [hunk 1]
```diff
@@ -713,6 +713,9 @@ parse_lone_slash = invalid trailing slash in literal
 parse_unskipped_whitespace = non-ASCII whitespace symbol '{$ch}' is not skipped
```
Comment (petrochenkov) [inline compiler/rustc_parse/locales/en-US.ftl:713]:
```suggestion
parse_unskipped_whitespace = whitespace symbol '{$ch}' is not skipped
```
I don't think the ASCII-ness matters here in any way?
What matters is that it is 1) whitespace and 2) not in the skipped set https://github.com/rust-lang/rust/pull/108403#issuecomment-1459144787.

So I think we can just change the error message here without changing any code.

File: compiler/rustc_parse/locales/en-US.ftl [hunk 2]
```diff
@@ -710,9 +710,12 @@ parse_zero_chars = empty character literal
 parse_lone_slash = invalid trailing slash in literal
     .label = {parse_lone_slash}
 
-parse_unskipped_whitespace = non-ASCII whitespace symbol '{$ch}' is not skipped
+parse_unskipped_whitespace = whitespace symbol '{$ch}' is not skipped
```
Comment (petrochenkov) [inline compiler/rustc_parse/locales/en-US.ftl:713]: I meant none of the changes except this line (and tests) are necessary, `UnskippedAsciiWhitespaceWarning` and `UnskippedAsciiWhitespace` can be removed.

Comment (bvanjoi) [reply compiler/rustc_parse/locales/en-US.ftl:713]: sorry for the misinterpretation...

**Pull request merged**